### PR TITLE
Data Table: Bot Accounts

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
Note: Due to the size of this current PR, the following will be branched out and fixed in separate PRs:
- Removal of new PAT panel upon bot deletion and PAT revoke

This PR refactors the bot account pages (Group and Project) to use the `data_table_component` (includes PAT tables to use `data_table_component`). This required much of the response logic (turbo etc.) to be re-done. 

## Screenshots or screen recordings
image

image

## How to set up and validate locally
Test all bot functionality:
1. Create new bot (table updates with new bot and new PAT panel appears/updates)
2. Delete bot (table updates, empty state appears if there are 0 bots)
3. Active PATs for a bot and Revoke action (empty state of PAT table once all PATs are revoked)
4. Generate new PAT (table active PAT # updates, new PAT panel appears/updates)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by joshsadam — CHANGES_REQUESTED** [review]
This looks really great overall.  I actually only have one wording change I would like to see.

File: config/locales/en.yml [hunk 1]
```diff
@@ -311,6 +311,34 @@ en:
...
+          level_0: No Access
+          level_10: Guest
+          level_20: Uploader
+          level_30: Analyst
+          level_40: Maintainer
+          level_50: Owner
+        actions:
+          generate_new_token: Generate new token
+          generate_new_token_aria_label: Generate a new personal access token for bot account
+          remove: Delete bot
```
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply config/locales/en.yml]: Agreed, changed in 90e8c7ed61effaa857e6a5de0cb6f0ef6097b8e2.

I've also changed the key from `remove` to `destroy` as that lines up more with what we've done for other translations.

**Review by deepsidhu85 — DISMISSED** [review]
Looks good to me!

**Comment by joshsadam** [comment]
I found a minor UI bug that I am hoping you can fix, and I am hoping that I can explain it properly. I will paste the screenshot and try to explain below.
bots

- The div highlighted in the developer console wraps a `turbo-frame`.
- When the turbo frame is empty (as in the screenshot) the div is empty but still renders and has the class of `mb-4` which for all purposes makes the margin bottom on the header 8.
- Can you remove the wrapping div and set the `mb-4` on the content within the `turbo-frame` so that it is only there if the `turbo-frame` has content?

**Comment by joshsadam** [comment]
I think this small bug is something that you can quickly fix in this PR.  The create  bot modal body does not vertically scroll.  If you don't have enough vertical screen space you cannot fill in all the values.  As yo can see in the screenshot I cannot select a date 
image

**Review by joshsadam — CHANGES_REQUESTED** [review]
This looks good.  I left 2 comments that I am hoping you can quickly fix. Good work on this.

**Comment by ChrisHuynh333** [comment]
> I found a minor UI bug that I am hoping you can fix, and I am hoping that I can explain it properly. I will paste the screenshot and try to explain below. bots
> 
> * The div highlighted in the developer console wraps a `turbo-frame`.
> * When the turbo frame is empty (as in the screenshot) the div is empty but still renders and has the class of `mb-4` which for all purposes makes the margin bottom on the header 8.
> * Can you remove the wrapping div and set the `mb-4` on the content within the `turbo-frame` so that it is only there if the `turbo-frame` has content?

Thanks, removed the wrapper and moved the `mb-4` to the frame content for both projects and groups in f039c402c4192177e1202278e122a8123f25ffa6

**Review by joshsadam — APPROVED** [review]
This looks good to me

**Pull request merged**